June Roundup (#111)
memfault/interrupt: _posts/2020-06-30-june-2020-roundup.md

---
title: "What we've been reading in June"
author: tyler
tags: [roundup]
---

<!-- excerpt start -->

Here are the articles, videos, and tools that we've been excited about this
June.

<!-- excerpt end -->

We hope you enjoy these links, and we look forward to hearing what you've been
reading in the comments.

## Articles & Learning

- [Embedded Systems Security and TrustZone](https://embeddedsecurity.io/) by
  Dimitrios Slamaris<br>Incredible resource published today on security best
  practices on an ARM Cortex-M, but starts from the basics with an intro into C
  programming, GNU tools, Cortex-M architecture, and bare-metal firmware
  tutorials. Really, really good. Bookmark this.
- [Fast Execute-Only Memory for Embedded Systems](https://arxiv.org/abs/2006.00076)

>   by Zhuojia Shen and John Criswell of Cornell University<br>A paper about

Reviewer: ```suggestion
  by Zhuojia Shen and John Criswell.<br>A paper about
```

Remove for formatting reason

  preventing remote code disclosure attacks with a clever use of the ARM
  Cortex-M MPU and DWT to implement Execute-Only-Memory (XOM).

> - nRF52 Debug Resurrection (APPROTECT Bypass) -

Reviewer: "A complete compromise of the flash readout protection on nRF52 series microcontrollers using simple voltage glitching techniques"

  [Part 1](https://limitedresults.com/2020/06/nrf52-debug-resurrection-approtect-bypass/)
  &
  [Part 2](https://limitedresults.com/2020/06/nrf52-debug-resurrection-approtect-bypass-part-2/)
  by LimitedResults<br>
- [Using ASCII waveforms to test hardware designs](https://blog.janestreet.com/using-ascii-waveforms-to-test-hardware-designs/)
  by Andrew Ray<br>A seriously cool approach to using "snapshot" based testing
  to verify waveforms. This particular approach converts a hardware waveform
  from the device-under-test to an ASCII representation and then compares that
  to a previously recorded value.
- [Bare metal RISC-V programming in Go](https://embeddedgo.github.io/2020/05/31/bare_metal_programming_risc-v_in_go.html)
  by Michał Derkacz<br>Continuing on the quest for running Go on embedded
  devices, this post documents the effort of porting the project to the RISC-V
  architecture.
- [Using Callbacks with Interrupts](https://www.beningo.com/using-callbacks-with-interrupts)
  by Jacob Beningo<br>A simple post about how and why to implement drivers using
  callbacks and interrupts.

> - [Making an Ultrasonic Anemometer](http://embedded-lab.com/blog/making-an-ultrasonic-anemometer/) -
>   Fascinating rabbit-hole in making a wind speed and direction sensor. The two
>   projects linked ([one](https://www.dl1glh.de/ultrasonic-anemometer.html),

Reviewer: ```suggestion
- [Making an Ultrasonic Anemometer](http://embedded-lab.com/blog/making-an-ultrasonic-anemometer/) by Raj Bhatt  <br>Fascinating rabbit-hole in making a wind speed and direction sensor. The two
  projects linked ([one](https://www.dl1glh.de/ultrasonic-anemometer.html),
```

  [two](http://soldernerd.com/arduino-ultrasonic-anemometer/)) are both
  individual engineers documenting the process of building their own anemometer.

## Neat Open Source Projects

- [F´ (pronounced F-Prime)](https://nasa.github.io/fprime/) by NASA<br>A
  software framework from NASA to assist with the development of spaceflight
  devices.

> - [Espressif RainMaker](https://rainmaker.espressif.com/) - I missed the initial
>   announcement, but it looks like the team at Espressif came out with an OTA
>   solution for their devices using a mobile-based provisioning technique.

Reviewer: ```suggestion
- [Espressif RainMaker](https://rainmaker.espressif.com/) - The team at Espressif came out with an OTA
  solution for their devices using a mobile-based provisioning technique. We're excited to see more chip vendors provide value-added software alongside their silicon!
```
